pkg/auth/dependency/authnsession/provider_impl.go

```diff
@@ -236,7 +236,12 @@ func (p *providerImpl) WriteResponse(w http.ResponseWriter, resp interface{}, er
 	if err == nil {
 		switch v := resp.(type) {
 		case model.AuthResponse:
-			p.sessionWriter.WriteSession(w, &v.AccessToken, &v.MFABearerToken)
+			// Do not touch the cookie if it is not in the response.
+			if v.MFABearerToken == "" {
```

Comment:
It seems we can't distinguish the cases of token is absent and token should be deleted. Do we have a way to delete a bearer token in cookie?

Reply:
Addressed this issue in commit `Return InvalidMFABearerToken and clear the token when it is invalid`